src/target/riscv/riscv-013.c

```diff
@@ -1175,6 +1175,12 @@ static int examine(struct target *target)
 		 * In order to make this work we first need to */
 		int offset = (progbuf_addr % 8 == 0) ? -4 : 0;
 
+		/* This program uses a temporary register. If the core can not
+		 * execute 64 bit instruction, the original value of temporary
+		 * register will not be restored due to an exception. So we have to
+		 * restore it manually in that case. */
+		uint64_t s0 = riscv_get_register(target, GDB_REGNO_S0);
```

Comment:
does this solve the chicken-and-egg? Do we actually know how to 'riscv_get_register' before executing this function?